framework/di/Instance.php

```diff
@@ -107,7 +107,9 @@ public static function of($id)
      */
     public static function ensure($reference, $type = null, $container = null)
     {
-        if ($reference instanceof $type) {
+        if (is_object($reference) && $type === null) {
+            return $reference;
+        } elseif (is_object($reference) && $reference instanceof $type) {
```

Comment:
There's no need for `is_object($reference)`. `$reference instanceof $type` is enough.